### PR TITLE
feat: add idex v4

### PR DESCRIPTION
**Pull request by dileepfrog**
Name (to be shown on DefiLlama):

IDEX

Twitter Link: 

https://x.com/idexio

List of audit links if any: https://github.com/idexio/idex-contracts-ikon/tree/main/audits 

Website Link: 

​​https://idex.io/

Logo (High resolution, will be shown with rounded borders): 

https://drive.google.com/file/d/1p0Bs_ZhTa8co6qdi1EPatgZXKsJi3fe9/view?usp=drive_link

Current TVL:

Treasury Addresses (if the protocol has treasury)

Chain:
XCHAIN
ChainID 94524
RPC https://xchain-rpc.idex.io
Explorer https://xchain-explorer.idex.io

Coingecko ID (so your TVL can appear on Coingecko, leave empty if not listed): (https://api.coingecko.com/api/v3/coins/list)

Coinmarketcap ID (so your TVL can appear on Coinmarketcap, leave empty if not listed): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)

Short Description (to be shown on DefiLlama):

Established in 2017, IDEX is transforming decentralized trading as the premier omnichain perpetuals exchange. The platform combines industry-leading speed and performance with transparency and security to create an entirely new trading experience featuring one-click onboarding, slick mobile trading with IDEX Connect and gas-free settlement secured by Ethereum. 

Token address and ticker if any:
IDEX
0xB705268213D593B8FD88d3FDEFF93AFF5CbDcfAE

Category (full list at https://defillama.com/categories) *Please choose only one:

Derivatives

Oracle Provider(s): Specify the oracle(s) used (e.g., Chainlink, Band, API3, TWAP, etc.):

Pyth

Implementation Details: Briefly describe how the oracle is integrated into your project:

https://api-docs-v4.idex.io/#index-prices 

Documentation/Proof: Provide links to documentation or any other resources that verify the oracle's usage:

https://api-docs-v4.idex.io/#index-prices 
https://docs.idex.io/index-price-oracle-and-weights

forkedFrom (Does your project originate from another project):
N/A

methodology (what is being counted as tvl, how is tvl being calculated):

Github org/user (Optional, if your code is open source, we can track activity):

### CODE REVIEW

File: dexs/idex-v4/index.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  return {
+    totalVolume: totalVolume?.toString(),
+    dailyVolume: volume24h?.toString(),
+  };
+};
+
+const adapter: SimpleAdapter = {
+  adapter: {
+    "idex-v4": {
```
Comment (dtmkeng) [inline dexs/idex-v4/index.ts]: chain name should `idex`  ?

**Review by dtmkeng — CHANGES_REQUESTED** [review]

**Comment by llamatester** [comment]
The idex-v4 adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts dexs idex-v4
    
    🦙 Running IDEX-V4 adapter 🦙
    ---------------------------------------------------
    Start Date:	Tue, 01 Oct 2024 00:00:00 GMT
    End Date:	Wed, 02 Oct 2024 00:00:00 GMT
    ---------------------------------------------------
    
    XCHAIN 👇
    Backfill start time: 18/7/2024
    NO METHODOLOGY SPECIFIED
    Total volume: 57.22 M
    Daily volume: 2.94 M

File: dexs/idex-v4/index.ts (hunk 1 above)
Comment (dileepfrog) [reply dexs/idex-v4/index.ts]: I misunderstood this key - the chain name is XCHAIN, the exchange name running there is IDEX

Comment (0xngmi) [reply dexs/idex-v4/index.ts]: xchain will likely get confused with the multiple other xchains there are, such as avalanche's, imo better to go with idex

Comment (dileepfrog) [reply dexs/idex-v4/index.ts]: Updated

**Comment by llamatester** [comment]
The idex-v4 adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts dexs idex-v4
    
    🦙 Running IDEX-V4 adapter 🦙
    ---------------------------------------------------
    Start Date:	Thu, 03 Oct 2024 00:00:00 GMT
    End Date:	Fri, 04 Oct 2024 00:00:00 GMT
    ---------------------------------------------------
    
    IDEX 👇
    Backfill start time: 18/7/2024
    NO METHODOLOGY SPECIFIED
    Total volume: 60.18 M
    Daily volume: 2.62 M

**Pull request merged**